traffic_ops/testing/api/assert/assertions.go

```diff
@@ -0,0 +1,144 @@
+		}
+		message = fmt.Sprintf("%+v", msg)
+	}
+	if len(msgAndArgs) > 1 {
+		message = fmt.Sprintf(msgAndArgs[0].(string), msgAndArgs[1:]...)
+	}
+	if len(message) > 0 {
+		output = output + " Messages: " + message
+	}
+	return output
+}
+
+// Equal asserts that two objects are equal.
+func Equal(t *testing.T, a, b interface{}, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if a != b {
+		msg := failureOutput(fmt.Sprintf("Error: Not equal. Expected: %v Actual: %v", a, b), msgAndArgs...)
+		t.Error(msg)
+		return false
+	}
+	return true
+}
+
+// RequireEqual asserts that two objects are equal.
+// It marks the test as failed and stops execution.
+func RequireEqual(t *testing.T, a, b interface{}, msgAndArgs ...interface{}) {
+	t.Helper()
+	if !Equal(t, a, b, msgAndArgs...) {
+		t.FailNow()
+	}
+}
+
+// Error asserts that a function returned an error (i.e. not `nil`).
+func Error(t *testing.T, err error, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if err == nil {
+		msg := failureOutput("Error: An error is expected but got nil.", msgAndArgs...)
+		t.Error(msg)
+		return false
+	}
+	return true
+}
+
+// Exactly asserts that two objects are equal in value and type.
+func Exactly(t *testing.T, a, b interface{}, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if !reflect.DeepEqual(a, b) {
+		msg := failureOutput(fmt.Sprintf("Error: Not exactly equal. Expected: %v Actual: %v", a, b), msgAndArgs...)
+		t.Error(msg)
+		return false
+	}
+	return true
+}
+
+// GreaterOrEqual asserts that the first element is greater than or equal to the second
+func GreaterOrEqual(t *testing.T, a, b int, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if a >= b {
+		return true
+	}
+	msg := failureOutput(fmt.Sprintf("Error: \"%v\" is not greater than or equal to \"%v\"", a, b), msgAndArgs...)
+	t.Error(msg)
+	return false
+}
+
+// RequireGreaterOrEqual asserts that the first element is greater than or equal to the second
+// It marks the test as failed and stops execution.
+func RequireGreaterOrEqual(t *testing.T, a, b int, msgAndArgs ...interface{}) {
+	t.Helper()
+	if !GreaterOrEqual(t, a, b, msgAndArgs...) {
+		t.FailNow()
+	}
+}
+
+// NoError asserts that a function returned no error (i.e. `nil`).
+func NoError(t *testing.T, err error, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if err != nil {
+		msg := failureOutput(fmt.Sprintf("Received unexpected error: %+v", err), msgAndArgs...)
+		t.Error(msg)
+		return false
+	}
+	return true
+}
+
+// RequireNoError asserts that a function returned no error (i.e. `nil`).
+// It marks the test as failed and stops execution.
+func RequireNoError(t *testing.T, err error, msgAndArgs ...interface{}) {
+	t.Helper()
+	if !NoError(t, err, msgAndArgs...) {
+		t.FailNow()
+	}
+}
+
+// NotNil asserts that the specified object is not nil.
+func NotNil(t *testing.T, a interface{}, msgAndArgs ...interface{}) bool {
+	t.Helper()
+	if a == nil {
+		msg := failureOutput("Error: Expected value not to be nil.", msgAndArgs...)
```

Comment:
nit: you don't need to add `Error:`, `testing.T.Error` provides the context of being a test error already.